### DISCUSSION TITLE
Sending mail to same domain, but not same server

### DISCUSSION THREAD

**Discussion by CelPlays** [docker-mailserver/docker-mailserver#3591] [General]
Hello!

I've setup the mail server for relaying emails from our various hosted services (sentry, gitlab, etc.) and I can succesfuly send emails to our non-domain emails such as potato@gmail.com, however when we try to send to hello@ourdomain.com, using noreply@ourdomain.com, we get

`550 5.1.1 <mail> User doesn't exist`

Which is true on the local server, but our mailboxes are hosted elsewhere (currently GSuite). We only use the mailserver to relay emails from our services to other emails (mostly our own domain emails).

Is it possible to just relay those emails without postfix attempting to send to a local user that doesn't exist? 

I setup the server with the [Using DMS as a local mail relay for containers](https://docker-mailserver.github.io/docker-mailserver/edge/examples/tutorials/basic-installation/#using-dms-as-a-local-mail-relay-for-containers) setup.

My aplogogies if this has already been asked, I looked for it but wasn't able to find it.

**Comment by leoborlot** [comment]
I think so, you need to disable the LMTP in DOVECOT https://doc.dovecot.org/configuration_manual/forwarding_parameters/

LMTP is a protocol that prioritizes email delivery within the same server if they belong to the same domain.

**Comment by polarathene** [comment]
> I setup the server with the [Using DMS as a local mail relay for containers](https://docker-mailserver.github.io/docker-mailserver/edge/examples/tutorials/basic-installation/#using-dms-as-a-local-mail-relay-for-containers) setup.

As noted that's a community guide, it's not been revised yet.

If you don't want DMS to think it owns `example.com`, then you should not configure it to own that domain.
- DNS MX record should point to the mail server that is responsible for delivering the mail to.
- DNS records like SPF should permit any other mail sending service you control to send mail on behalf of the domain. Which is what you're using DMS for?

Your other services should then use authenticated submission on ports 465 or 587 to send mail outbound. This requires an account created in DMS but that does not have to match the sender address (_only if using `SPOOF_PROTECTION=1`_), you can then send mail to an address as any sender address that suits you.

---

You can configure DMS as a null mailer with `SMTP_ONLY=1`, this will disable Dovecot that is used for mail storage, but it's also the default account provider.

Presently there is no fallback for managing users with `SMTP_ONLY=1`, only [community workarounds](https://github.com/docker-mailserver/docker-mailserver/issues/3477#issuecomment-1676500609).

This is probably more in line with what you want, to use DMS only for sending mail from your services, not to receive any mail or manage any real user mail accounts. This would be similar to services like SendGrid where mail is relayed through them to the real destination.

If you've configured for `mail.example.com`, you could also use `setup email add noreply@mail.example.com` and use that for the authentication, while sending the mail with `noreply@example.com`.

IIRC there are some legacy parts of the DMS scripts that [try to be helpful](https://github.com/docker-mailserver/docker-mailserver/issues/3520) by configuring some parts with `example.com` instead of `mail.example.com` 🤷‍♂️
